File: korge-spine/src/jvmMain/java/com/badlogic/gdx/files/FileHandle.java
package com.badlogic.gdx.files;

import kotlin.NotImplementedError;

import java.io.Reader;

public class FileHandle {
    public String nameWithoutExtension() {
        throw new NotImplementedError();
    }

    public byte[] read(int i) {
        return new byte[0];
    }

    public Reader reader(String charset) {
        throw new NotImplementedError();
    }

    public String pathWithoutExtension() {
        return null;
    }

    public String extension() {
        return null;
    }
}

File: korge-spine/src/jvmMain/kotlin/com/badlogic/gdx/files/FileHandle.kt
package com.badlogic.gdx.files

import java.io.*
import java.net.*

class FileHandle(val url: URL) {
    constructor(resourcePath: String, classLoader: ClassLoader = ClassLoader.getSystemClassLoader()) : this(
        classLoader.getResource(resourcePath)
    )

    fun nameWithoutExtension(): String {
        return url.file.substringBeforeLast('.')
    }

    fun read(@Suppress("UNUSED_PARAMETER") bufferSize: Int): ByteArray {
        return url.readBytes()
    }

    fun reader(charset: String?): Reader {
        TODO()
    }

    fun pathWithoutExtension(): String? {
        TODO()
    }

    fun extension(): String? {
        TODO()
    }
}
